src/components/Sections/GoogleMapsSection/index.tsx

```diff
@@ -0,0 +1,179 @@
+import { Component, Prop } from "vue-property-decorator";
+import BaseComponent from "@/components/BaseComponent";
+import {
+  GoogleMapsSectionProps,
+  MapsPosition,
+  MapsLocation,
+} from "@/types/sections";
+import { LineSeparator } from "@/components";
+import { Loader } from "@googlemaps/js-api-loader";
+// eslint-disable-next-line
+const markerIcon = require("../../../assets/Marker.png");
+
+const DEFAULT_STYLE: google.maps.MapTypeStyle[] = [
+  {
+    featureType: "water",
+    elementType: "geometry",
+    stylers: [{ color: "#e9e9e9" }, { lightness: 17 }],
+  },
+  {
+    featureType: "landscape",
+    elementType: "geometry",
+    stylers: [{ color: "#f5f5f5" }, { lightness: 20 }],
+  },
+  {
+    featureType: "road.highway",
+    elementType: "geometry.fill",
+    stylers: [{ color: "#ffffff" }, { lightness: 17 }],
+  },
+  {
+    featureType: "road.highway",
+    elementType: "geometry.stroke",
+    stylers: [{ color: "#ffffff" }, { lightness: 29 }, { weight: 0.2 }],
+  },
+  {
+    featureType: "road.arterial",
+    elementType: "geometry",
+    stylers: [{ color: "#ffffff" }, { lightness: 18 }],
+  },
+  {
+    featureType: "road.local",
+    elementType: "geometry",
+    stylers: [{ color: "#ffffff" }, { lightness: 16 }],
+  },
+  {
+    featureType: "poi",
+    elementType: "geometry",
+    stylers: [{ color: "#f5f5f5" }, { lightness: 21 }],
+  },
+  {
+    featureType: "poi.park",
+    elementType: "geometry",
+    stylers: [{ color: "#dedede" }, { lightness: 21 }],
+  },
+  {
+    elementType: "labels.text.stroke",
+    stylers: [{ visibility: "on" }, { color: "#ffffff" }, { lightness: 16 }],
+  },
+  {
+    elementType: "labels.text.fill",
+    stylers: [{ saturation: 36 }, { color: "#333333" }, { lightness: 40 }],
+  },
+  { elementType: "labels.icon", stylers: [{ visibility: "off" }] },
+  {
+    featureType: "transit",
+    elementType: "geometry",
+    stylers: [{ color: "#f2f2f2" }, { lightness: 19 }],
+  },
+  {
+    featureType: "administrative",
+    elementType: "geometry.fill",
+    stylers: [{ color: "#fefefe" }, { lightness: 20 }],
+  },
+  {
+    featureType: "administrative",
+    elementType: "geometry.stroke",
+    stylers: [{ color: "#fefefe" }, { lightness: 17 }, { weight: 1.2 }],
+  },
+] as google.maps.MapTypeStyle[];
+
+@Component({
+  name: "GoogleMapsSection",
+})
+class GoogleMapsSection extends BaseComponent<GoogleMapsSectionProps> {
+  map: google.maps.Map | null = null;
+
+  @Prop() title: GoogleMapsSectionProps["title"];
+  @Prop({ required: false }) apikey!: GoogleMapsSectionProps["apikey"];
```

Comment:
The `apiKey` should be required since this section won't work if no (valid) `apiKey` is provided.